include/deal.II/grid/manifold_lib.h

```diff
@@ -503,6 +503,150 @@ class TorusManifold : public ChartManifold<dim,3,3>
+   * and computes the new point by a transfinite interpolation. The first step
+   * in the implementation is to identify the coarse cell on which the
+   * surrounding points are located. Then, the coordinates are transformed to
+   * the unit coordinates on the coarse cell by a Newton iteration, where the
+   * new point is then computed according to the weights. Finally, it is
+   * pushed forward to the real space according to the transfinite
+   * interpolation.
+   */
+  virtual
+  Point<spacedim>
+  get_new_point (const std::vector<Point<spacedim> > &surrounding_points,
+                 const std::vector<double>           &weights) const;
+
+  /**
+   * Compute a new set of points that interpolate between the given points
+   * @p surrounding_points. @p weights is a table with as many columns as
+   * @p surrounding_points.size(). The number of rows in @p weights determines
+   * how many new points will be computed and appended to the last input
+   * argument @p new_points. After exit of this function, the size of
+   * @p new_points equals the size at entry plus the number of rows in
+   * @p weights.
+   *
+   * The implementation in this class overrides the method in the base class
+   * and computes the new point by a transfinite interpolation. The first step
+   * in the implementation is to identify the coarse cell on which the
+   * surrounding points are located. Then, the coordinates are transformed to
+   * the unit coordinates on the coarse cell by a Newton iteration, where the
+   * new points are then computed according to the weights. Finally, the is
+   * pushed forward to the real space according to the transfinite
+   * interpolation.
+   *
+   * The implementation does not allow for @p surrounding_points and
+   * @p new_points to point to the same vector, so make sure to pass different
+   * objects into the function.
+   */
+  virtual
+  void
+  add_new_points (const std::vector<Point<spacedim> > &surrounding_points,
+                  const Table<2,double>               &weights,
+                  std::vector<Point<spacedim> >       &new_points) const;
+
+private:
+  /**
+   * Internal function to identify the cell where the given surrounding points
+   * are localted.
+   */
+  typename Triangulation<dim,spacedim>::cell_iterator
+  get_cell_around_points(const std::vector<Point<spacedim> > &surrounding_points) const;
+
+  /**
+   * Pull back operation into the unit coordinates on the given coarse cell.
+   *
+   * @note This internal function is not compatible with the
+   * ChartManifold::pull_back() function because the given class represents
+   * an atlas of charts, not a single chart. Thus, the pull_back() operation
+   * is only valid with the additional information of the chart, given by a
+   * cell on the coarse grid.
+   */
+  Point<dim>
+  pull_back(const typename Triangulation<dim,spacedim>::cell_iterator &cell,
+            const Point<spacedim> &p) const;
```

Comment:
I don't think we can derive from a `ChartManifold`. While I could call `pull_back` that simply finds the point around the ones given as you suggest, I can not reasonably define the `push_forward` function because that function **must** know which of the charts to pick from the atlas. Just from the unit coordinates there is no way to tell. Unless I'm overlooking something here, I do not see why to expose those functions or wrap them into the interface of `ChartManifold`.